src/common/sharedlibrary/shared_library.cpp

```diff
@@ -56,6 +57,13 @@ void* mir::SharedLibrary::load_symbol(char const* function_name) const
 
 void* mir::SharedLibrary::load_symbol(char const* function_name, char const* version) const
 {
+    // Some libc implementations (such as musl) do not support dlvsym
+
+#ifdef DONT_USE_DLVSYM
+    // Load the function without checking the version
+    log_debug("Cannot check %s symbol version is %d: dlvsym() is unavailable", function_name, version);
+    return load_symbol(function_name);
+#else
```

Comment:
I still don't like using the preprocesor - as I've said elsewhere it stops the compiler validating the code. (But I won't block on this as I've stopped caring.)

Reply:
A CI based on Alpine Linux should probably be added once the tests are working fine to avoid regressions (and then also this code piece gets validated).